lib/experimental/full-page-client-side-navigation.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Registers full page client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+// Set the navigation mode to full page client-side navigation.
+wp_interactivity_config( 'core/router', array( 'navigationMode' => 'fullPage' ) );
```

Comment:
Should this code be colocated with enqueuing the router? It also only applies to the front end for the time being.

Reply:
Done as part of [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/181614e418fa958f90077d7a3e9a409326126299).